git-fixup

```diff
@@ -46,116 +55,130 @@ function fixup_candidates_lines () {
 # staged changes
-function fixup_candidates_files () {
+fixup_candidates_files () {
     git diff --cached --name-only | (
-        while read file; do
-            git rev-list $rev_range -- $file \
-                | grep -v -f <(git rev-list -E --grep='^(fixup|squash)' $rev_range -- $file) \
+        TMP_COMMIT_LIST="$(mktemp --tmpdir gitfixup-XXXX)"
+        trap 'rm -f "$TMP_COMMIT_LIST"' EXIT
+        while read -r file ; do
+            git rev-list -E --grep='^(fixup|squash)' "$rev_range" -- "$file" >"$TMP_COMMIT_LIST"
+            git rev-list "$rev_range" -- "$file" \
+                | grep -v -f "$TMP_COMMIT_LIST" \
                 | head -n1
         done
     ) | sed 's/^/F /g'
 }
 
 # Produce suggestion of all commits in $rev_range
-function fixup_candidates_all_commits () {
-	git rev-list $rev_range | sed 's/^/F /g'
+fixup_candidates_all_commits () {
+	git rev-list "$rev_range" | sed 's/^/F /g'
 }
 
 # Pretty print details of a commit
-function print_sha () {
-    local sha=$1
-    local type=$2
+print_sha () {
+    sha=$1
+    type=$2
 
     git --no-pager log --format="%H [$type] %s <%ae>" -n 1 "$sha"
 }
 
 # Call git commit
-function call_commit() {
-    local flag=$op
-    local target=$1
+call_commit() {
+    set -x
+    flag=$op
+    target=$1
 
-    if test "$op" == "amend"; then
+    if [ "$op" = "amend" ] ; then
         flag=fixup
         target="amend:$target"
     fi
 
-    git commit ${git_commit_args[@]} --$flag=$target || die
+    # shellcheck disable=SC2086
+    git commit ${git_commit_args} "--$flag=$target" || die
 }
 
 # Call git rebase
-function call_rebase() {
-    local target=$1
+call_rebase() {
+    target=$1
 
     # If our target-commit has a parent, we call a rebase with that
-    if git rev-parse --quiet --verify $target~1^{commit}; then
+    # shellcheck disable=SC1083
+    if git rev-parse --quiet --verify "$target"~1^{commit} ; then
         git rebase --interactive --autosquash "$target~1"
     # If our target-commit exists but has no parents, it must be the very first commit
     # the repo. We simply call a rebase with --root
-    elif git rev-parse --quiet --verify $target^{commit}; then
+    elif git rev-parse --quiet --verify "$target"^{commit} ; then
         git rebase --interactive --autosquash --root
     fi
 }
 
 # Print list of fixup/squash candidates
-function print_candidates() {
+print_candidates() {
     (
-        if test "$show_all" == "false"; then
+        if [ "$show_all" = "false" ] ; then
             fixup_candidates_lines
             fixup_candidates_files
         else
             fixup_candidates_all_commits
         fi
-    ) | sort -uk2 |  while read type sha; do
-        if test "$sha" != ""; then
+    ) | sort -uk2 | tee /tmp/git-fixup-log-$$.txt | while read -r type sha ; do
+        if [ -n "$sha" ] ; then
             print_sha "$sha" "$type"
         fi
     done
 }
 
-function fallback_menu() {
+fallback_menu() {
```

Comment:
I'm of half a mind to remove the fallback menu already and just have a message to configure fzf (or something like it) if a menu is desired. It adds a lot of noise to the script and even more in good ol' posix shell.